mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/api/index.ts

```diff
@@ -0,0 +1,69 @@
+import apiFetch from '@wordpress/api-fetch';
+import { dispatch, select } from '@wordpress/data';
+import { getCurrentDates } from '@woocommerce/date';
+import { api } from '../config';
+import { storeName } from '../store/constants';
+import { Query, Section, SectionData } from '../store/types';
+
+export type ApiError = {
+  code?: string;
+  message?: string;
+  data?: {
+    status?: number;
+    details?: Error;
+    params?: Record<string, string>;
+    errors?: unknown[];
+  };
+};
+
+export const initializeApi = () => {
+  const apiUrl = `${api.root}/mailpoet/v1/`;
+  apiFetch.use(apiFetch.createRootURLMiddleware(apiUrl));
+  apiFetch.use(apiFetch.createNonceMiddleware(api.nonce));
+};
+
+export async function updateSection(
```

Comment:
You seem to be using a Redux store. Why not make this method a Redux action?

Reply:
I am not sure. Not really sure about best practice when it comes to Redux but it felt "bloated" to me. Currently what `updateSection` does:

1. Fetch the current query and automation from the store
2. Transform it to an API query
3. Empty the current section data in the store
4. Run the API query
5. Populate the current section data in the store

So it felt to me like it should be it's own function interacting with the store API (through dispatch etc) instead of being an specialized "action", which to me feels like it should only be concerned with the store itself. Moving this into an action feels to me like

```
class DBLayer {
  update() {
    //fetch data from somewhere
    //transform data
    mysqli_query($this->connection, "UPDATE ...");
  }
}
```

Reply:
@websupporter I think calling API and processing the responses is what's done also in other MailPoet and WP data stores. [WP even has utilities for that](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data-controls/), and if you want to extract the logic from the action, then WP has [controls](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#controls) for that (you can see them used in the form editor). Although they can be a bit more writing (apart from an action and a reducer, you need to implement a control as well), they can separate some concerns, if you want to separate them.

I kind of get what you mean, but maybe with actions and controls, "everything belongs to Redux". I'd prefer not to invent an additional custom data layer when `@wordpress/data` is likely meant to handle that, at least to my understanding.